packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -283,6 +290,60 @@ class _PosixUtils extends OperatingSystemUtils {
   }
 }
 
+class _LinuxUtils extends _PosixUtils {
+  _LinuxUtils({
+    required FileSystem fileSystem,
+    required Logger logger,
+    required Platform platform,
+    required ProcessManager processManager,
+  }) : super(
+          fileSystem: fileSystem,
+          logger: logger,
+          platform: platform,
+          processManager: processManager,
+        );
+
+  String? _name;
+
+  @override
+  String get name {
+    if (_name == null) {
+      const String prettyNameKey = 'PRETTY_NAME';
+      final String osReleasePath = _fileSystem.file('/etc/os-release').existsSync()
+        ? '/etc/os-release'
+        : '/usr/lib/os-release';
+      try {
+        final String osRelease = _fileSystem.file(osReleasePath).readAsStringSync();
+        final String prettyName = _getOsReleaseValueForKey(osRelease, prettyNameKey);
+        final String kernelRelease = _platform.operatingSystemVersion.split(' ')[1];
+        _name = '$prettyName$kernelRelease';
+      } on Exception catch (e) {
+        _logger.printTrace('Failed obtaining name for Linux: $e');
```

Comment:
I'd suggest restructuring this a bit to be a bit less all-or-nothing. E.g.:
1. Try to get PRETTY_NAME, and if it fails omit it from the result or use something generic like "Linux"
2. Try to get _platform.operatingSystemVersion, and if it fails omit it from the result
3. If both (1) and (2) fail, fall back on super.name.

Reply:
Since `super.name` is generic itself, we could probably check whether `PRETTY_NAME` is empty, either due to a failed attempt to read `os-release` or `PRETTY_NAME` being missing(although it is very unlikely), and replace it with `super.name`:
```dart
String prettyName;
String kernelRelease;
try {
  final String osRelease = _fileSystem.file(osReleasePath).readAsStringSync();
  prettyName = _getOsReleaseValueForKey(osRelease, prettyNameKey);
} on Exception catch (e) {
  _logger.printTrace('Failed obtaining PRETTY_NAME for Linux: $e');
  prettyName = '';
}
try {
  kernelRelease = ' ${_platform.operatingSystemVersion.split(' ')[1]}';
} on Exception catch (e) {
  _logger.printTrace('Failed obtaining kernel release for Linux: $e');
  kernelRelease = '';
}
_name = '${prettyName.isEmpty ? super.name : prettyname}$kernelRelease'
```
What do you think?